docs/sc-memory/api/cpp/agents.md

```diff
@@ -0,0 +1,450 @@
+3. Create file `CMakeLists.txt` with content (replace `my_module` (module name) to your one):
+
+**CMakeLists.txt**
+
+```bash
+# Define alias for directory path with your sc-module and sc-agents in it.
+set(MY_AGENT_SRC ${CMAKE_CURRENT_LIST_DIR})
+
+# Collect your sc-agent sources.
+set(SOURCES
+    "my_module.cpp"
+)
+set(HEADERS
+	"my_module.hpp"
+)
+
+# Include using other directories.
+include_directories(${MY_AGENT_SRC} ${SC_MEMORY_SRC})
+
+# Create and link your library with using libraries.
+add_library(my_lib SHARED ${SOURCES} ${HEADERS})
+add_dependencies(my_lib sc-memory)
+target_link_libraries(my_lib sc-memory)
+
+# Specify where code for your sc-agent will be generated.
+sc_codegen_ex(my_lib ${MY_AGENT_SRC} "${MY_AGENT_SRC}/generated")
+```
+
+After that you need to create two files (.cpp and .hpp):
+
+**my_module.hpp**
+
+```cpp
+#pragma once
+
+#include <sc-memory/sc_module.hpp>
+
+#include "my_module.generated.hpp"
+
+// Your module class for sc-agents
+class MyModule : public ScModule
+{
+  // Priority that can be used during load all sc-modules.
+  SC_CLASS(LoadOrder(10))
+  SC_GENERATED_BODY()
+
+  virtual sc_result InitializeImpl() override;
+  virtual sc_result ShutdownImpl() override;
+}
+```
+
+**my_module.cpp**
+
+```cpp
+#include "my_module.hpp"
+
+SC_IMPLEMENT_MODULE(MyModule)
+
+// This method will be called once. 
+// It is usually used to register sc-agents and sc-keynodes.
+sc_result MyModule::InitializeImpl()
+{
+  return SC_RESULT_OK;
+}
+
+// This method will be called once. 
+// It is usually used to unregister sc-agents and sc-keynodes.
+sc_result MyModule::ShutdownImpl()
+{
+  return SC_RESULT_OK;
+}
+```
+
+## **ScAgent**
+
+To create sc-agent implementation you need:
+
+1. Create source and header file for sc-agent implementation.
+2. Add created files into `CMakeLists.txt`.
+3. Register implemented sc-agent in created sc-module.
+
+**CMakeLists.txt**
+
+```diff
+...
+
+set(SOURCES
+  "my_module.cpp"
++ "my_agent.cpp"
+)
+set(HEADERS
+  "my_module.hpp"
++ "my_agent.hpp"
+)
+
+...
+```
+
+**my_agent.hpp**
+
+```cpp
+#pragma once
+
+#include "sc-memory/kpm/sc_agent.hpp"
+#include "sc-memory/sc_keynodes.hpp"
+
+#include "my_agent.generated.hpp"
+
+// Use namespaces to user can import in his code only needed sc-agents.
+namespace myAgents
+{
+
+class MyAgent : public ScAgent
+{
+  // Specify sc-event on which this sc-agent will be called.
+  SC_CLASS(
+      Agent, 
+      Event(ScKeynodes::question_initiated, ScEvent::Type::AddOutputEdge)
+  )
+  // This sc-agent can be called if sc-connector is created from sc-element 
+  // with sc-address `ScKeynodes::question_initiated`.
+  SC_GENERATED_BODY()
+}
+
+}
+```
+
+**my_agent.cpp**
+
+```cpp
+#include "my_agent.hpp"
+
+#include "sc-memory/sc_memory.hpp"
+
+namespace myAgents
+{
+
+SC_AGENT_IMPLEMENTATION(MyAgent)
+{
+  // Implement sc-agent logic here. 
+  // Use Core C++ API to implement logic of your sc-agent.
+  return SC_RESULT_OK;
```

Comment:
It is feature of interpreter of scp-interpreter. We can call agents by initiation sc-events of output connectors adding from their action classes. I propose to use in example `my_action_class` instead of `question_initiated`.